### PR TITLE
Fix Latest Clippy Lints for object_store

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 

Follow up to #5544

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: object_store/src/aws/client.rs [hunk 1]
```diff
@@ -138,6 +138,7 @@ struct BatchDeleteResponse {
 
 #[derive(Deserialize)]
 enum DeleteObjectResult {
+    #[allow(unused)]
```
Comment (tustvold) [inline object_store/src/aws/client.rs:141]: It was complaining because we never use this field outside of the deserialization routine

**Comment by tustvold** [comment]
Merging as uncontroversial and unblocks CI

File: object_store/src/aws/client.rs (hunk 1 above)
Comment (alamb) [reply object_store/src/aws/client.rs:141]: so is it actually unused? Or is this a clippy false positive?

**Review by alamb — COMMENTED** [review]
Thanks @tustvold

**Pull request merged**